### PR TITLE
Feature/#69 create cfei modal

### PR DESCRIPTION
**Pull request by marcindo**
Added modals for creating new cfei/direct selection
- added fileForm field
- added better loader logic

### CODE REVIEW

File: frontend/src/components/common/modals/controlledModal.js [hunk 1]
```diff
@@ -20,13 +16,13 @@ const styleSheet = createStyleSheet('ContentDialog', theme => ({
   },
 }));
 
-function infoSection(info, classes) {
-  return info ? <div className={classes.info}><DialogContent>{info}</DialogContent></div> : null;
-}
+const messages = {
+  cancel: 'Cancel',
+  ok: 'Ok',
```
Comment (kenzo23312) [inline frontend/src/components/common/modals/controlledModal.js]: empty line

File: frontend/src/components/eois/cells/eoiCountryCell.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import withCountryName from '../../common/hoc/withCountryName';
+
+const eoiCountryCell = (props) => {
+  const { countryName } = props;
+  return (
+    <div>
+      {countryName}
```
[the remark this answers is not in the document]
Comment (marcindo) [reply frontend/src/components/eois/cells/eoiCountryCell.js]: From what I can see it doesn't change anything, I believe text in table cell is already styled properly

File: frontend/src/components/eois/eoiPartnerTable.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  { id: 'country_code', label: 'Country' },
+  { id: 'sector', label: 'Sector' },
+  { id: 'agency', label: 'Agency' },
+  { id: 'deadline_date', label: 'Application deadline' },
+  { id: 'start_date', label: 'Project start date' },
+];
+
+export const renderCells = (item, classes) => [
+  <TableCell className={`${classes.limitedCell} ${classes.firstCell}`}>
+    {item.title}
```
[the remark this answers is not in the document]
Comment (marcindo) [reply frontend/src/components/eois/eoiPartnerTable.js]: From what I can see it doesn't change anything, I believe text in table cell is already styled properly

File: frontend/src/components/eois/modals/callPartnersModal.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  onDialogSubmit() {
+    this.props.submit();
+  }
+
+  onDialogClose() {
+    this.props.newCfeiProcessed();
+  }
+
+
```
[the remark this answers is not in the document]
Comment (marcindo) [reply frontend/src/components/eois/modals/callPartnersModal.js]: done

File: frontend/src/components/eois/eoiPartnerTable.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            data={cfei}
+            columnData={columnData}
+            title={messages.title}
+            renderTableCells={renderCells}
+            loading={loading}
+          />
+        </Grid>
+        <AlertDialog
+          trigger={alert}
+          title="Warning"
```
[the remark this answers is not in the document]
Comment (marcindo) [reply frontend/src/components/eois/eoiPartnerTable.js]: done

File: frontend/src/components/common/table/sharedTable.js [hunk 6]
```diff
@@ -20,17 +18,36 @@ class SharedTable extends Component {
...
   }
 
+  componentWillReceiveProps(nextProps) {
+    return this.setState({ data: nextProps.data });
+  }
+
   handleRequestSort(event, property) {
     const orderBy = property;
+    let sortingFunc;
     let order = 'desc';
```
[the remark this answers is not in the document]
Comment (marcindo) [reply frontend/src/components/common/table/sharedTable.js]: done

**Pull request merged**